src/Confluent.Kafka/AdminClient.cs

```diff
@@ -0,0 +1,750 @@
+        ///
+        ///     - allTopics = true - request all topics from cluster
+        ///     - allTopics = false - request only locally known topics.
+        /// 
+        ///     [API-SUBJECT-TO-CHANGE] - The API associated with this functionality is subject to change.
+        /// </summary>
+        public Metadata GetMetadata(bool allTopics, TimeSpan timeout)
+            => kafkaHandle.GetMetadata(allTopics, null, timeout.TotalMillisecondsAsInt());
+
+
+        /// <summary>
+        ///     Query the cluster for metadata (blocking).
+        ///
+        ///     - allTopics = true - request all topics from cluster
+        ///     - allTopics = false, topic = null - request only locally known topics.
+        ///     - allTopics = false, topic = valid - request specific topic
+        /// 
+        ///     [API-SUBJECT-TO-CHANGE] - The API associated with this functionality is subject to change.
+        /// </summary>
+        public Metadata GetMetadata(bool allTopics, string topic, TimeSpan timeout)
+            => kafkaHandle.GetMetadata(
+                allTopics, 
+                topic == null ? null : getKafkaTopicHandle(topic), 
+                timeout.TotalMillisecondsAsInt());
+
+
+        /// <summary>
+        ///     Refer to <see cref="Confluent.Kafka.IClient.AddBrokers(string)" />
+        /// </summary>
+        public int AddBrokers(string brokers)
+            => kafkaHandle.AddBrokers(brokers);
+
+
+        /// <summary>
+        ///     Refer to <see cref="Confluent.Kafka.IClient.Name" />
+        /// </summary>
+        public string Name
+            => kafkaHandle.Name;
+
+
+        /// <summary>
+        ///     An opaque reference to the underlying librdkafka 
+        ///     client instance.
+        /// </summary>
+        public Handle Handle
+            => handle;
+
+
+        /// <summary>
+        ///     Releases all resources used by this AdminClient. In the current
+        ///     implementation, this method may block for up to 100ms. This 
+        ///     will be replaced with a non-blocking version in the future.
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+
+        /// <summary>
+        ///     Releases the unmanaged resources used by the
+        ///     <see cref="Confluent.Kafka.AdminClient" />
+        ///     and optionally disposes the managed resources.
+        /// </summary>
+        /// <param name="disposing">
+        ///     true to release both managed and unmanaged resources;
+        ///     false to release only unmanaged resources.
+        /// </param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                callbackCts.Cancel();
+                try
+                {
+                    callbackTask.Wait();
+                }
+                catch (AggregateException e)
```

Comment:
that's a great question because `Dispose` should never throw. I believe there is no other possible case here, if another exception happens it's an application logic error. I'll put a comment stating that.